pyext/src/output.py

```diff
@@ -692,59 +692,148 @@ def write_stats2(self):
+
+        if header:
+            # Assume that STAT2HEADER_ENVIRON is the last keyword
+            if "STAT2HEADER_ENVIRON" in line:
+                d['STAT2HEADER_ENVIRON']=line.split(", \'STAT2HEADER_ENVIRON\':")[1]
+                line=line.split(", \'STAT2HEADER_ENVIRON\'")[0]
+
+        # First, remove outer braces and split via double quotes to isolate multi-component values.
+        split=line.strip()[1:-1].split("\"")
+        
+        if len(split)==1:
+            fields = split[0].split(",")   # split via commas to get key:value pair
+            for h in fields: 
+                if h != "":  # For some reason, there is occasionally an empty field. Ignoring these seems to work.
+
+                    # Split fields into key and value elements
+                    kv = h.split(":")
+
+                    # If the key (field 0) is an integer, keep it an integer.
+                    try:
+                        k = int(kv[0].replace(",","").strip())
+                    except:
+                        k = kv[0].replace("\'","").strip()
+
+                    v = kv[1].replace("\'","").strip() # the value is encased in single quote characters, so remove these
+                    d[k] = v
+            return d
+
+
+        for i in range(0,len(split),2):
+            # Each even number of split is a string of "key:value, key:value, key:value"
+            fd = split[i]
+            fields = fd.split(",")   # split via commas to get key:value pair
+            for h in fields[0:-1]: 
+                if h != "":  # For some reason, there is occasionally an empty field. Ignoring these seems to work.
+                    #print(h, h.split(": "))
+
+                    # Fields are separated by ': '
+                    kv = h.split(": ")
+
+                    # If the key (field 0) is an integer, keep it an integer.
+                    try:
+                        k = int(kv[0].replace(",","").replace(":","").strip())
+                    except:
+                        k = kv[0].replace("\'","").strip()
+
+                    v = kv[1].replace("\'","").strip() # the value is encased in single quote characters, so remove these
+                    d[k] = v
+
+            # If there is a last field contains the key for the multi-component value in split[i+1]
+            if i < len(split)-1:
+                d[fields[-1].split(":")[0].replace("\'","").strip()] = split[i+1]
+            else:
+                kv = fields[-1].split(": ")
+                try:
+                    k = int(kv[0].replace(",","").replace(":","").strip())
+                except:
+                    k = kv[0].replace("\'","").strip()
+
+                v = kv[1].replace("\'","").strip() # the value is encased in single quote characters, so remove these
+                d[k] = v
+
+        return d
+
+
     def get_keys(self):
+        """ Returns a list of the string keys that are included in this dictionary
+        """
+        self.klist = [k[1]
```

Comment:
Do this only once, not each time the function is called. Easiest would just be to setup `klist` when the file is first parsed.